### PR TITLE
Add __truediv__ to simulate `pathlib.Path`

### PR DESCRIPTION
**Pull request by moonknows**
Add __truediv__ to simulate pathlib.Path

### CODE REVIEW

**Comment by Yevgnen** [comment]
Thanks for contributing. 

I think it's reasonable to remove `__call__` since `/` is more intuitive. Keeping both may lead to confusion. Do you mind adjusting the changes?

**Comment by moonknows** [comment]
No problem.

File: carton/file.py [hunk 1]
```diff
@@ -27,7 +27,11 @@ def __call__(self, *paths):
     def norm(self):
         return self.__class__(normalize_path(self))
 
+    def __truediv__(self, *paths):
```
Comment (Yevgnen) [inline carton/file.py]: Based on the [document](https://docs.python.org/3/reference/datamodel.html#object.__truediv__) of Python 3, I think `__truediv__` should accept exactly single argument.

**Comment by moonknows** [comment]
屏幕快照 2021-08-29 16 04 17

File: carton/file.py [hunk 2]
```diff
@@ -19,15 +19,14 @@ class _Path(str):
...
-    def __call__(self, *paths):
-        paths = (self,) + tuple(paths)
-
-        return _Path(os.path.join(*paths))
-
     def norm(self):
         return self.__class__(normalize_path(self))
 
+    def __truediv__(self, path):
+        path = f"{str(path).lstrip('/')}"
```
Comment (Yevgnen) [inline carton/file.py]: I think you don't need the `lstrip` call here...

File: carton/file.py [hunk 3]
```diff
@@ -19,15 +19,13 @@ class _Path(str):
...
-
-        return _Path(os.path.join(*paths))
-
     def norm(self):
         return self.__class__(normalize_path(self))
 
+    def __truediv__(self, path):
+        return _Path(os.path.join(self, path))
 
+
```
Comment (Yevgnen) [inline carton/file.py]: Please clean up the empty spaces...

**Comment by Yevgnen** [comment]
LGTM, thanks!

**Comment by moonknows** [comment]
Love you!!!

**Pull request merged**